### PR TITLE
Improve code coverage on api/gin

### PR DESCRIPTION
**Pull request by sstanculeanu**

### CODE REVIEW

File: api/gin/httpServer_test.go [hunk 1]
```diff
@@ -1,17 +1,91 @@
...
+		t.Parallel()
+
+		hs, err := NewHttpServer(nil)
+		assert.Equal(t, errors.ErrNilHttpServer, err)
+		assert.True(t, check.IfNil(hs))
+	})
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		hs, err := NewHttpServer(&http.Server{})
```
Comment (iulianpascalau) [inline api/gin/httpServer_test.go:26]: maybe use a test server?
valid on all occurrences

File: api/gin/httpServer_test.go [hunk 2]
```diff
@@ -1,17 +1,91 @@
...
+
+		server := &http.Server{
+			Addr: "127.0.0.1:8080",
+		}
+		hs, _ := NewHttpServer(server)
+		assert.False(t, check.IfNil(hs))
+
+		go hs.Start()
+		time.Sleep(3 * time.Second)
+		err := hs.Close()
```
Comment (iulianpascalau) [inline api/gin/httpServer_test.go:88]: this might fail as the 8080 might already be in use. Suggestion to use a test server

**Review by iulianpascalau — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

File: api/gin/httpServer.go [hunk 3]
```diff
@@ -9,11 +9,11 @@ import (
 )
 
 type httpServer struct {
-	server *http.Server
```
Comment (iulianpascalau) [inline api/gin/httpServer.go:12]: please change https://github.com/ElrondNetwork/elrond-eth-bridge/blob/7c86f6528dcf6362918df17f91f64cd184c7bf52/.github/workflows/golangci-lint.yml#L19
to `v1.45` in order to make the golang ci linter pass

File: testsCommon/facade/relayerfacadeStub.go [hunk 4]
```diff
@@ -41,7 +41,7 @@ func (stub *RelayerFacadeStub) RestApiInterface() string {
 // PprofEnabled -
 func (stub *RelayerFacadeStub) PprofEnabled() bool {
 	if stub.PprofEnabledCalled != nil {
-		stub.PprofEnabledCalled()
+		return stub.PprofEnabledCalled()
```
Comment (dragos-rebegea) [inline testsCommon/facade/relayerfacadeStub.go:44]: :+1:

**Review by dragos-rebegea — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**